fili-core/src/main/java/com/yahoo/bard/webservice/logging/RequestLog.java

```diff
@@ -401,6 +401,18 @@ public static String getId() {
     }
 
     /**
+     * Set id prefix.
+     *
+     * @param idPrefix  Prefix for queryId sent to druid
+     */
+    public static void addIdPrefix(String idPrefix) {
+        RequestLog current = RLOG.get();
+        String newId = idPrefix + getId();
+        current.logId = newId;
+        MDC.put(ID_KEY, newId);
```

Comment:
While this will update the `current.logId` field (and the MDC value), I _don't_ think it will update the UUID that is part of the `RequestLog` payload. I _think_ that the `LogInfo.uuid` field (on the object stored in `current.info`) needs to be updated as well in order for that to happen.

I think the reason this was missed is because we're holding that UUID in 2 places `current.logId` (we read from that all over), and in the `LogInfo.uuid` field (`current.info`). It would be _great_ if we could get rid of the `current.logId` field and only use the value from the `LogInfo` object. This refactor isn't required for this PR, but it would be great if we could pull it in sooner, rather than later.

As part of the change around where the logId is accessed from, we should update `getId` to pull from the single source of truth as well.

Also, since this is the closest I can get to it, it would be cool (if we do this refactoring) to move the `MDC.put` call in the `init` method up much higher, so that there's less of a gap between getting a UUID and setting it into MCD.